src/vdaf.rs

```diff
@@ -415,18 +415,24 @@ where
         .map(|input_share| input_share.get_encoded());
 
     let mut states = Vec::new();
+    let mut outbound = Vec::new();
     for (verify_param, input_share) in verify_params.iter().zip(input_shares) {
-        let state = vdaf.prepare_init(
+        let (state, msg) = vdaf.prepare_init(
             verify_param,
             agg_param,
             nonce,
             &V::InputShare::get_decoded_with_param(verify_param, &input_share)
                 .expect("failed to decode input share"),
         )?;
         states.push(state);
+        outbound.push(msg.get_encoded());
     }
 
-    let mut inbound = None;
+    let mut inbound = vdaf.prepare_preprocess(outbound.iter().map(|encoded| {
+        V::PrepareMessage::get_decoded_with_param(&states[0], encoded)
+            .expect("failed to decode papare message")
```

Comment:
```suggestion
            .expect("failed to decode prepare message")
```
But in any case, why not propagate the error? I think something like
```
let mut inbound = vdaf.prepare_preprocess(outbound.iter().map(|encoded| V::PrepareMessage::get_decoded_with_param(...)).transpose().collect::<Vec<Result<_, _>>>())
```
might work.